python/tank/api.py

```diff
@@ -183,7 +184,7 @@ def shotgun_url(self):
         """
         The associated shotgun url, e.g. ``https://mysite.shotgunstudio.com``
         """
-        return shotgun.get_associated_sg_base_url()
+        return core.CoreSettings().host
```

Comment:
I haven't managed to find a way to simply use `CoreSettings().host` and still be able to mock part of the API for unit tests. I'm not really happy about this actually.